From 7b47cc93075dfb7c93f40d07f15d7efa19351e6d Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 10 Sep 2013 11:29:52 +0900
Subject: [PATCH 1/2] ext/dfbvideosink: invoke the flush method after clearing
 surfaces

This fix is required to use the clear method from different threads,
which are the streaming thread and the state change thread.
The flush method is used to flush the internal call buffer of DirectFB.

ATTENTION:
  The flush method is the new API of DirectFB in version 1.8.0.
  The following commit of DirectFB is a must to use this patch.

commit b5301bba8c0878d86cb5a1901ccc587ecea20673
"API: Add IDirectFBSurface::Flush() to flush the internal call buffer as done implicitly by Flip()"
---
 ext/directfb/dfbvideosink.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index cd43472..9c59b3d 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -215,6 +215,9 @@ gst_dfbvideosink_clear_surface (GstDfbVideoSink * dfbvideosink,
         dfbvideosink->bgcolor.green, dfbvideosink->bgcolor.blue,
         dfbvideosink->bgcolor.alpha);
   }
+
+  surface->Flush (surface);
+
   dfbvideosink->require_clear_surface = 0;
   g_mutex_unlock (dfbvideosink->window_lock);
 }
-- 
1.8.1.2

